template/provider.py

```diff
@@ -481,7 +481,7 @@ def __init__(self, params):
 
     if isinstance(paths, str):
       paths = re.split(dlim, paths)
-    if size == 1 or size < 0:
+    if size == None:
```

Comment:
I think the right fix here is to keep this line unchanged, but change `size = params.get("CACHE_SIZE")` to `size = params.get("CACHE_SIZE", -1)